core/processor/tx.go

```diff
@@ -42,13 +43,19 @@ func DecodeTxNoValidation(payload []byte) (*Tx, error) {
 	if err := err.ErrorOrNil(); err != nil {
 		return nil, err
 	}
+	ttl := uint64(0)
+	// TTL block only applies to TXv4
+	if tx.Version == commandspb.TxVersion_TX_VERSION_V4 {
+		ttl = inputData.BlockHeight + inputData.GoodForBlocks
+	}
 
 	return &Tx{
 		originalTx: payload,
 		tx:         tx,
 		inputData:  inputData,
 		pow:        tx.Pow,
 		version:    tx.Version,
+		ttl:        ttl,
```

Comment:
Why do we need to save the tll separately and not just do:
```
func (t Tx) TTL() uint64 {
	return t.inputData.BlockHeight + t.inputData.GoodForBlocks
}
```

then in `checkTx` we just simply have:
```
if tx.GetVersion() > 3 && tx.Command().IsTTLCommand() {
    // do thing
}
```
because we also avoid having the equality here `tx.Version == commandspb.TxVersion_TX_VERSION_V4` which someone will no doubt forget to update when we roll into tx-version 5.

Reply:
So we're saving TTL separately because the TTL depends not just on the `GoodForBlocks` part of the TX, but a network parameter. The max TTL also doubles as the default for TxV3, and it overrides any value set here if the `GoodForBlocks` >  `MaxTTL`. So when decoding the TX, we check the netparam, and set the TTL accordingly. That being said, I suspect the `TX_VERSION_V4` check now can probably be replaced with a `if tx.inputData.GoodForBlocks > 0 && tx.inputData.GoodForBlocks < maxTTL`.

Reply:
Makes sense now that its a network-parameter.